delta/src/main/scala/io/qbeast/spark/delta/DeltaRollupDataWriter.scala

```diff
@@ -40,23 +39,24 @@ import java.net.URI
  */
 object DeltaRollupDataWriter extends RollupDataWriter with DeltaStatsCollectionUtils {
 
-  override type GetCubeMaxWeight = CubeId => Weight
-  override type Extract = InternalRow => (InternalRow, Weight, CubeId, CubeId)
-  override type WriteRows = Iterator[InternalRow] => Iterator[(IndexFile, TaskStats)]
-
   override def write(
       tableId: QTableID,
       schema: StructType,
       data: DataFrame,
       tableChanges: TableChanges): IISeq[IndexFile] = {
+
+    if (data.isEmpty) return Seq.empty[IndexFile].toIndexedSeq
+
     val revision = tableChanges.updatedRevision
     val dimensionCount = revision.transformations.length
 
     val statsTrackers = StatsTracker.getStatsTrackers
     val fileStatsTracker = getFileStatsTracker(tableId, data)
     val trackers = statsTrackers ++ fileStatsTracker
 
-    val filesAndStats = internalWrite(tableId, schema, data, tableChanges, trackers)
+    val extendedData = extendDataWithCubeToRollup(data, tableChanges)
+    val filesAndStats = doWrite(tableId, schema, extendedData, tableChanges, trackers)
```

Comment:
where's the internalWrite call?

Reply:
Has been renamed to `doWrite()`, in the `RollupDataWriter`